File: 113-path-sum-ii/113-path-sum-ii.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> pathSum(TreeNode* root, int targetSum) {
        auto ans=get(root,targetSum);
        for(auto &val:ans){
            reverse(val.begin(),val.end());
        }
        return ans;
    }
    vector<vector<int>> get(TreeNode *root,int target){
            vector<vector<int>> ans;
        if(root){
            if(!root->left && !root->right){
                if(root->val==target){
                    vector<int> t={root->val};
                    ans.push_back(t);
                }
                return ans;
            }
            ans=get(root->left,target-root->val);
            for(auto &val:ans){
                val.push_back(root->val);
            }
            auto a=get(root->right,target-root->val);
            for(auto &val:a){
                val.push_back(root->val);
                ans.push_back(val);
            }
            return ans;
        }
        return ans;
    }
};

File: 113-path-sum-ii/113-path-sum-ii.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def pathSum(self, root: Optional[TreeNode], targetSum: int) -> List[List[int]]:
        ans=self.get(root,targetSum)
        for item in ans:
            item.reverse()
        return ans
    def get(self,root,target):
        if(root):
            if(not root.left and not root.right):
                if(root.val==target):
                    return [[root.val]]
                return []
            ans=self.get(root.left,target-root.val)
            ans+=self.get(root.right,target-root.val)
            for item in ans:
                item.append(root.val)
            return ans
        return []